pkg/registry/core/service/storage/storage.go

```diff
@@ -56,12 +58,19 @@ func NewGenericREST(optsGetter generic.RESTOptionsGetter, serviceCIDR net.IPNet,
 	}
 	options := &generic.StoreOptions{RESTOptions: optsGetter}
 	if err := store.CompleteWithOptions(options); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	statusStore := *store
 	statusStore.UpdateStrategy = service.NewServiceStatusStrategy(strategy)
-	return &GenericREST{store}, &StatusREST{store: &statusStore}, nil
+	// Returning a function such that the caller will be able to customize the AfterDelete
+	// step for the service resource as necessary. AfterDelete should implement a further
+	// operation to run after a service is deleted and before it is decorated.
```

Comment:
I can't find specific public document around this AfterDelete operation. The closest is the comment from the registry.Store field. So I put something similar here.
https://github.com/kubernetes/kubernetes/blob/09cd56b561d0c31421dfc6374b68e0b18c6cc746/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L163-L165